test/test_torch.py

```diff
@@ -6152,7 +6152,8 @@ def run(model0, model1, optimizer0, optimizer1, try_scaling_api):
     @onlyNativeDeviceTypes
     def test_grad_scaler_pass_itself(self, device):
         device = torch.device(device)
-        GradScaler = torch.cuda.amp.GradScaler if "cuda" == device.type else torch.cpu.amp.GradScaler
+        device_type = "cuda" if "cuda" == device.type else "cpu"
+        GradScaler = partial(torch.amp.GradScaler, device=device_type)
```

Comment:
```suggestion
        GradScaler = partial(torch.amp.GradScaler, device=device)
```
should work?

Reply:
here defines `onlyNativeDeviceTypes`
https://github.com/pytorch/pytorch/blob/86ad101370075cfa392c953b17e1a8ab3e66881f/torch/testing/_internal/common_utils.py#L230
For `meta` and `PU1` we should test it on `cpu`.

Reply:
following your suggestion, update the code.